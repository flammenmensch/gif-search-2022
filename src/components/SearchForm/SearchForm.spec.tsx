import { fireEvent, render, screen } from '@testing-library/react';
import { SearchForm } from './index';

test('It renders', () => {
  render(<SearchForm />);
  const form = screen.getByRole(/^search$/);
  expect(form).toBeInTheDocument();
});

test('It matches snapshot', () => {
  const { asFragment } = render(<SearchForm />);
  expect(asFragment()).toMatchSnapshot();
});

test('It receives focus on render', () => {
  render(<SearchForm />);
  const searchBox = screen.getByRole(/searchbox/);

  expect(searchBox).toHaveFocus();
});

test('It handles submit', () => {
  const handler = jest.fn();
  render(<SearchForm onSearch={handler} />);
  const form = screen.getByRole(/^search$/);
  const searchBox = screen.getByRole(/searchbox/);
  const query = 'query';

  fireEvent.change(searchBox, { target: { value: query } });
  fireEvent.submit(form);

  expect(handler).toHaveBeenCalledTimes(1);
  expect(handler).toHaveBeenCalledWith(query);
});
