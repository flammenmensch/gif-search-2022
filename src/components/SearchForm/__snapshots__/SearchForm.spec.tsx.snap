// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It matches snapshot 1`] = `
<DocumentFragment>
  <form
    role="search"
  >
    <div
      class="stz-core-floating-label"
    >
      <input
        id=":r1:"
        placeholder="Search query"
        role="searchbox"
        type="text"
        value=""
      />
      <label
        for=":r1:"
      >
        Search query
      </label>
      <small>
        e.g. kittens
      </small>
    </div>
  </form>
</DocumentFragment>
`;
